experimental/federated_plugin/federated_plugin_web/test_driver/federated_plugin_web_e2e.dart

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:html';
+
+import 'package:e2e/e2e.dart';
+import 'package:federated_plugin_web/federated_plugin_web.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+const kLatitude = 4;
+const kLongitude = 3;
+
+class WindowMock extends Mock implements Window {
+  @override
+  Navigator get navigator => NavigatorMock();
+}
+
+class NavigatorMock extends Mock implements Navigator {
+  @override
+  Geolocation get geolocation => GeoLocationMock();
+}
+
+class GeoLocationMock extends Mock implements Geolocation {
+  @override
+  Future<Geoposition> getCurrentPosition(
+      {bool enableHighAccuracy, Duration timeout, Duration maximumAge}) async {
+    return GeoPositionMock();
+  }
+}
+
+class GeoPositionMock extends Mock implements Geoposition {
+  @override
+  Coordinates get coords => MockCoordinates();
+}
+
+class MockCoordinates extends Mock implements Coordinates {
+  @override
+  num get latitude => kLatitude;
+
+  @override
+  num get longitude => kLongitude;
+}
```

Comment:
@RedBrogdon Instead of mocking `FederatedPlugin` I decided to mock all the other classes to provide concrete test since it also test the conversation of `html.Coordinates` to `Location` for `getLocation`.

Reply:
I have improved the implementation of plugin and test to reduce lines of code.